lib/marked.js

```diff
@@ -14,10 +14,9 @@
 var block = {
   newline: /^\n+/,
   code: /^( {4}[^\n]+\n*)+/,
-  fences: noop,
+  fences: /^ {0,3}(`{3,}|~{3,})([^`\n]*)\n(?:|([\s\S]*?)\n)(?: {0,3}\1[~`]* *(?:\n+|$)|$)/,
```

Comment:
This is quadratic due to the sub-regex `...(`{3,}|~{3,})([^`\n]*)...` <-- note that the second group does not exclude `~`. An attack string of the form `~~...~` will parse slowly.

Reply:
I just copied that from gfm.fences. So it doesn't make marked any less secure.

Could I fix it just by adding a `~` to the second group?

Reply:
Yes, that would be linear.